File: c/soln02.c
#include <stdio.h>

int main() {
    // Formula: Xn+1 = Xn + C/Xn
    float X0 = 2;
    float C = 5;
    printf("Formula: Xn+1 = Xn + C/Xn\n");
    printf("X0: %f\n", X0);

    for (int n = 1; n < 12; n++) {
        float Xn = (X0 + (C / X0)) / 2;
        if (Xn == X0) {
            break;
        } else {
            printf("X%d", n, ": %9.6f\n", Xn);
            X0 = Xn;
        }
    }

    return 0;
}

File: java/soln02.java
package java;
public class soln02 {

    private static void println(Object object) {
        System.out.println(object);
    }

    public static void main(String[] args) {
        // Formula: Xn+1 = Xn + C/Xn

        double X0 = 2;
        double C = 5;

        println("Formula: Xn+1 = Xn + C/Xn\n");
        println("X0: " + X0);
        for (int n = 1; n < 11; n++) {
            double Xn = (X0 + (C /X0))/2;
            if (Xn == X0) {
                break;
            } else {
                println("X"+ n + ": " + (double) Math.round(Xn * 1000000) / 1000000D);
                X0 = Xn;
            }
        }
    }
}
